src/compiler/parser.ts

```diff
@@ -5391,7 +5391,7 @@ namespace ts {
             // Avoiding having to do the lookahead for a labeled statement by just trying to parse
             // out an expression, seeing if it is identifier and then seeing if it is followed by
             // a colon.
-            const node = <ExpressionStatement | LabeledStatement>createNodeWithJSDoc(SyntaxKind.Unknown);
+            const node = <ExpressionStatement | LabeledStatement>createNodeWithJSDoc(token() === SyntaxKind.Identifier ? SyntaxKind.Unknown : SyntaxKind.ExpressionStatement);

```

Comment:
What other things can token be here? Why is it that only `Identifier` leaves the node's kind as unknown?

Reply:
This change is necessary so we can correctly handle `ExpressionStatement` inside `createNodeWithJSDoc`. You are right, that it would've been set later anyway, but that's too late as the JSDoc would already be attached to the `ExpressionStatement` node.

`Identifier` has a special handling here, because it's the only token that could be the start of a `LabeledStatement`. This change is only about `ExpressionStatement`s starting with parentheses. Hence we can leave this as `Unknown` and avoid the lookahead to determine whether we are actually dealing with a `LabeledStatement`.